homeassistant/components/bmw_connected_drive/coordinator.py

```diff
@@ -53,9 +53,20 @@ async def _async_update_data(self) -> None:
 
         try:
             await self.account.get_vehicles()
-        except (HTTPError, TimeoutException) as err:
-            self._update_config_entry_refresh_token(None)
-            raise UpdateFailed(f"Error communicating with BMW API: {err}") from err
+        except (HTTPError, HTTPStatusError, TimeoutException) as err:
+            if isinstance(err, HTTPStatusError) and err.response.status_code == 429:
+                _LOGGER.warning(
```

Comment:
Why shouldn't we raise `UpdateFailed`? We could lower the update interval on the coordinator if we get a 429 error and still raise UpdateFailed. Logging a warning will spam the log with warnings.

Reply:
Yes, sensors become unavailable. That's expected if an update fails.

If we change the coordinator update interval it will take effect the next time a refresh is scheduled. We know that a refresh will be scheduled after each refresh as long as there are listeners and authentication is valid.

So we can set a different interval on a specific error and reset it when the update succeeds again.

https://github.com/home-assistant/core/blob/9d132521424804cfd482229524124f3ab5742123/homeassistant/helpers/update_coordinator.py#L280-L281

Reply:
As the underlying data from the vehicle doesn't change that much and the users rather like to have old data than none, I'm keeping the Warning instead of `UpdateFailed`.
I have however implemented an increased polling period.

Reply:
No, we shouldn't spam the log with warnings. Home Assistant should make the entities unavailable in this case. That's standard.